cli/lib/nf3.mjs

```diff
@@ -1309,19 +1307,64 @@ class Nf3 {
   }
 
   /**
-    Send offchain transaction to Optimist
+    Intended to be called by a Proposer acting as a Proxy for offchain transactions. This
+    function allows the Proposer to forward offchain transactions to its Optimist instance.
     @method
     @async
-    @param {string} transaction
-    @returns {array} A promise that resolves to the API call status
-    */
-  async sendOffchainTransaction(transaction) {
-    const res = axios.post(
+    @param {object} transaction
+    @returns {object} A promise that resolves to the API call status
+   */
+  async forwardOffchainTransaction(transaction) {
+    const res = await axios.post(
       `${this.optimistBaseUrl}/proposer/offchain-transaction`,
       { transaction },
       { timeout: 3600000 },
     );
-    return res.status;
+    return res;
+  }
+
+  /**
+    Send offchain transaction to all Proposers that the chain knows about
+    This is intended to be called by a User to send the transaction to a Proposer
+    who will act as a proxy and forward the transaction to an Optimist (this allows an
+    Optimist to be firewalled off and only the Proposer exposed to Users).
+    @method
+    @async
+    @param {object} transaction
+    @returns {object} A promise that resolves to the API call status
+    */
+  async sendOffchainTransaction(transaction) {
+    // we're proxying offchain transactions through the proposers. We send the transaction to every proposer
+    // dig up connection peers
+    const currentProposer = await this.stateContract.methods.currentProposer().call();
+    const peerList = { [currentProposer.thisAddress]: currentProposer.url };
+    let nextProposer = await this.stateContract.methods
+      .proposers(currentProposer.nextAddress)
+      .call();
+    let proposerIdx = 0;
+    while (
+      currentProposer.thisAddress !== nextProposer.thisAddress &&
+      proposerIdx <= this.NEXT_N_PROPOSERS
+    ) {
+      peerList[nextProposer.thisAddress] = nextProposer.url;
+      // eslint-disable-next-line no-await-in-loop
+      nextProposer = await this.stateContract.methods.proposers(nextProposer.nextAddress).call();
+      proposerIdx += 1;
+    }
+
+    logger.debug({ msg: 'Peer List', peerList });
+    return Promise.any(
+      Object.keys(peerList).map(async address => {
+        logger.debug(
+          `offchain transaction - calling ${peerList[address]}/proposer/offchain-transaction`,
+        );
+        return axios.post(
```

Comment:
I think you can use the function `forwardOffchainTransaction` here.

Reply:
I don't think so because that would assume that the URL which is put on chain is the same as the URL of the Optimist.  This is not the case if you are proxying through the Proposer.